components/calendar.tsx

```diff
@@ -0,0 +1,469 @@
+              date.getFullYear() === today.getFullYear();
+
+            const isWeekend = date.getDay() === 0 || date.getDay() === 6;
+
+            return (
+              <li
+                key={date.toISOString()}
+                ref={(el) => {
+                  if (listRef?.current) {
+                    listRef.current[idx] = el;
+                  }
+                }}
+                role="option"
+                aria-selected={isHighlighted}
+                id={`option-${idx}`}
+                onClick={() => {
+                  if (isWeekend && disableWeekend) {
+                    return;
+                  }
+                  handleSelect(date);
+                }}
+                onMouseEnter={() => setHighlightedIndexChange(idx)}
+                tabIndex={isHighlighted ? 0 : -1}
+                className={cn(
+                  `flex self-center justify-center text-center`,
+                  isHighlighted &&
+                    (!disableWeekend || !isWeekend) &&
+                    "cursor-pointer"
+                )}
+              >
+                <span
+                  className={cn(
+                    "w-6 h-6 rounded-full relative border border-transparent",
+                    floatingStyles ? "p-[3px]" : "p-[2px]",
+                    disableWeekend && isWeekend
+                      ? "text-gray-300"
+                      : isWeekend && "text-red-300",
+                    isHighlighted
+                      ? "border-[#61A9F9] border text-[#61A9F9]"
+                      : "hover:bg-blue-200 focus:outline-none focus:bg-blue-200",
+                    isHighlighted &&
+                      disableWeekend &&
+                      isWeekend &&
+                      "bg-white text-gray-300 border-transparent select-none cursor-default",
+                    isCurrentDate
+                      ? "bg-[#61A9F9] text-white"
+                      : isToday && !isCurrentDate
+                        ? "text-[#61A9F9]"
+                        : isToday && isHighlighted && "text-white"
+                  )}
+                >
+                  {date.getDate()}
+                  {isToday && (
+                    <div className="border-[#61A9F9] bg-[#61A9F9] absolute bottom-[1px] left-[10px] border w-[3px] h-[3px]"></div>
+                  )}
+                </span>
+              </li>
+            );
+          })}
+        </div>
+      </>
+    </ul>
+  );
+}
+
+function formatDate(date: Date, format: FormatProps) {
+  const year = date.getFullYear();
+  const month = String(date.getMonth() + 1).padStart(2, "0");
+  const day = String(date.getDate()).padStart(2, "0");
+
+  switch (format) {
+    case "yyyy-mm-dd":
+      return `${year}-${month}-${day}`;
+    case "dd/mm/yyyy":
+      return `${day}/${month}/${year}`;
+    case "mm/dd/yyyy":
+    default:
+      return `${month}/${day}/${year}`;
+  }
+}
+
+const Calendar = CalendarDrawer;
+export { Calendar, CalendarDrawer };
```

Comment:
let's not export CalendarDrawer here, since it's not a drawer; that other components think of it as a drawer is not the business of this component :)